### PR TITLE
Add matching strategy demo

### PR DESCRIPTION
**Pull request by CaroFG**

### CODE REVIEW

**Comment by CaroFG** [comment]
Need to update instant-meilisearch to a version compatible with Meilisearch v0.29

**Review by mdubus — COMMENTED** [review]
I would have loved having a small explanation at the beginning of the readme, to know what to expect from this demo. Maybe this demo will come with a blog post explaining all of this, but just by reviewing it I was lost and had to try a few things and search inside the specs of the core to find what was the `matchingStrategy` and understand what it was.
Maybe users will find this demo by just browsing our repositories, so I think a little sentence explaining the aim of this demo would be great (even better if you provide a small example of the search results to see the difference between the two settings) 🙏

**Comment by mdubus** [comment]
Can we also have a bit of style for this? 🙏

Capture d’écran 2022-09-22 à 08 33 17

**Comment by CaroFG** [comment]
I added more info on the README with a link to the documentation. I'll update the link once the new documentation is live

File: src/matching-strategy/setup/setupFunctions.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+async function sleep (ms) {
+  return await new Promise((resolve) => setTimeout(resolve, ms))
+}
+
+async function watchTasks (client, uid) {
+  let allProcessed = false
+  console.log(`Start update watch for ${uid}`)
+  console.log('-------------')
+  while (!allProcessed) {
```
Comment (mdubus) [inline src/matching-strategy/setup/setupFunctions.js:13]: Sorry I wasn't clear enough in my previous message 🙈 
You can also remove the `allProcessed` variable and the `while` loop 🤩

**Review by mdubus — APPROVED** [review]
LGTM 🎉

**Pull request merged**